test/Microsoft.Azure.Devices.Edge.Test.Common/Platform.cs

```diff
@@ -1,27 +1,116 @@
+                    Log.Verbose(string.Join("\n", output));
+                },
+                "Created certificates for edge device");
 
-        public static string GetConfigYamlPath() => IsWindows()
-            ? Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData) + @"\iotedge\config.yaml"
-            : "/etc/iotedge/config.yaml";
+            var files = new[]
+            {
+                $"certs/iot-edge-device-{deviceId}-full-chain.cert.pem",
+                $"private/iot-edge-device-{deviceId}.key.pem",
+                "certs/azure-iot-test-only.root.ca.cert.pem"
+            };
+
+            files = NormalizeFiles(files, scriptPath).ToArray();
+
+            return new EdgeCertificates(files[0], files[1], files[2]);
+        }
+
+        public static async Task<LeafCertificates> GenerateLeafCertificatesAsync(
+            string leafDeviceId,
+            string scriptPath,
+            (string name, string args) command,
+            CancellationToken token)
+        {
+            await Profiler.Run(
+                async () =>
+                {
+                    string[] output = await Process.RunAsync(command.name, command.args, token);
+                    Log.Verbose(string.Join("\n", output));
+                },
+                "Created certificates for leaf device");
+
+            var files = new[]
+            {
+                $"certs/iot-device-{leafDeviceId}-full-chain.cert.pem",
+                $"private/iot-device-{leafDeviceId}.key.pem"
+            };
+
+            files = NormalizeFiles(files, scriptPath).ToArray();
+
+            return new LeafCertificates(files[0], files[1]);
+        }
+
+        public static async Task InstallRootCertificateAsync(
+            string scriptPath,
+            (string name, string args) command,
+            CancellationToken token)
+        {
+            string[] output = await Process.RunAsync(command.name, command.args, token);
+            Log.Verbose(string.Join("\n", output));
+
+            var files = new[]
+            {
+                "certs/azure-iot-test-only.root.ca.cert.pem",
+                "private/azure-iot-test-only.root.ca.key.pem"
+            };
+
+            CheckFiles(files, scriptPath);
+        }
+
+        public static void InstallTrustedCertificates(IEnumerable<X509Certificate2> certs, StoreName storeName)
+        {
+            using (var store = new X509Store(storeName, StoreLocation.CurrentUser))
+            {
+                store.Open(OpenFlags.ReadWrite);
+                foreach (var cert in certs)
+                {
+                    store.Add(cert);
+                }
+            }
+        }
 
         public static bool IsWindows() => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+
+        static void CheckFiles(IEnumerable<string> paths, string basePath) => NormalizeFiles(paths, basePath);
```

Comment:
why need this "CheckFiles" method? it calls NormalizeFiles only.  Need to find a better name for NormalizeFiles?

Reply:
NormalizeFiles actually normalizes file names, so that's the right name. CheckFiles reuses the code but discards the return value, so you can't use CheckFiles to normalize file names. The one caller could just call NormalizeFiles and discard the return value; I just thought this passthrough function made it clear that the return value was being discarded intentionally rather than by mistake.

Reply:
I found that when CheckFiles is called, paths.Select is not run as linq query is deferred execution.

Reply:
`
class Program
    {
        static void Main(string[] args)
        {
            test2();
        }

        public static void test2() => TestEnumerable();

        public static IEnumerable<int> TestEnumerable()
        {
            var values = new List<int>{0,1,2,3,4,5,6,7};
            return values.Select(v =>
            {
                Console.WriteLine(v);
                return v;
            });
        }
    }
`

Reply:
> I found that when CheckFiles is called, paths.Select is not run as linq query is deferred execution.

Ah, interesting. I guess that makes sense. I'll fix it.

Reply:
OK. All callers of `NormalizeFiles` (except `CheckFiles`) where immediately calling `ToArray` anyway, so I just pushed that into the method and changed its return type to `string[]`.